gr-fec/python/fec/qa_ecc_ccsds_27.py

```diff
@@ -36,6 +36,20 @@ def xtest_ccsds_27(self):
         dst_data = dst.data()
         self.assertEqual(expected, dst_data)
 
+    def test_ccsds_27_long(self):
+        src_data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] * 100000
+        src = blocks.vector_source_b(src_data)
+        enc = fec.encode_ccsds_27_bb()
+        b2f = blocks.char_to_float()
+        add = blocks.add_const_ff(-0.5)
+        mul = blocks.multiply_const_ff(2.0)
+        dec = fec.decode_ccsds_27_fb()
+        dst = blocks.vector_sink_b()
+        self.tb.connect(src, enc, b2f, add, mul, dec, dst)
+        self.tb.run()
+        dst_data = dst.data()
+        self.assertEqual(src_data[-104:-4], dst_data[-100:])
```

Comment:
I'm only comparing the last 100 output bytes here, because `assertEqual` is very slow when comparing long lists.

Reply:
I had a second look at this, and `assertEqual` is actually quite fast in the success case (~40 ms) whereas `np.testing.assert_equal` takes more than 11 seconds. `assertEqual` only becomes slow when the lists are unequal, but that won't happen anyway unless one of the blocks under test is broken. I'll update this to compare the entire list.

Reply:
That's a bit surprising, but anyhow, since `assertEqual` is fast in the success case, the idea to use it with the full list is good. I suspect that in the failure case it spends too much time trying to find how the inputs differ and how to format the error output.

Reply:
The issue is discussed here: https://github.com/python/cpython/issues/63416

It looks like it may be fixed in Python 3.12.